cmd/gcs-sidecar/main.go

```diff
@@ -0,0 +1,252 @@
+	logFile := flag.String("logfile",
+		defaultLogFile,
+		"Logging Target. Default is at C:\\gcs-sidecar-logs.log inside UVM")
+	confidentialWCOW := flag.Bool("confidential",
+		false,
+		"If true, start gcs-sidecar for confidential windows containers")
+	initialPolicyStance := flag.String("initial-policy-stance",
+		"allow",
+		"Stance: allow, deny.")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\nUsage of %s:\n", os.Args[0])
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "Examples:\n")
+		fmt.Fprintf(os.Stderr, "    %s -loglevel=trace -logfile=C:\\sidecarLogs.log \n", os.Args[0])
+	}
+
+	flag.Parse()
+
+	ctx := context.Background()
+	logFileHandle, err := os.OpenFile(*logFile, os.O_RDWR|os.O_CREATE|os.O_SYNC|os.O_TRUNC, 0666)
+	if err != nil {
+		fmt.Printf("error opening file: %v", err)
+	}
+	defer logFileHandle.Close()
+
+	logrus.AddHook(shimlog.NewHook())
+
+	level, err := logrus.ParseLevel(*logLevel)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	logrus.SetLevel(level)
+	logrus.SetOutput(logFileHandle)
+	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+	trace.RegisterExporter(&oc.LogrusExporter{})
+
+	if err := windows.SetStdHandle(windows.STD_ERROR_HANDLE, windows.Handle(logFileHandle.Fd())); err != nil {
+		logrus.WithError(err).Error("error redirecting handle")
+		return
+	}
+	os.Stderr = logFileHandle
+
+	chsrv := make(chan error)
+	go func() {
+		defer close(chsrv)
+
+		if err := runService("gcs-sidecar", false); err != nil {
+			logrus.Fatalf("error starting gcs-sidecar service: %v", err)
+		}
+
+		chsrv <- err
+	}()
+
+	select {
+	case <-ctx.Done():
+		logrus.Fatalln("context deadline exceeded")
+		return
+	case r := <-chsrv:
+		if r != nil {
+			logrus.Fatal(r)
+			return
+		}
+	}
+
+	// 1. Start external server to connect with inbox GCS
+	listener, err := winio.ListenHvsock(&winio.HvsockAddr{
+		VMID:      prot.HvGUIDLoopback,
+		ServiceID: prot.WindowsGcsHvsockServiceID,
+	})
+	if err != nil {
+		logrus.WithError(err).Errorf("error starting listener for sidecar <-> inbox gcs communication")
+		return
+	}
+
+	var gcsListener net.Listener = listener
+	gcsCon, err := acceptAndClose(ctx, gcsListener)
+	if err != nil {
+		logrus.WithError(err).Errorf("error accepting inbox GCS connection")
+		return
+	}
+
+	// 2. Setup connection with external gcs connection started from hcsshim
+	hvsockAddr := &winio.HvsockAddr{
+		VMID:      prot.HvGUIDParent,
+		ServiceID: prot.WindowsSidecarGcsHvsockServiceID,
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"hvsockAddr": hvsockAddr,
+	}).Tracef("Dialing to hcsshim external bridge at address %v", hvsockAddr)
+	shimCon, err := winio.Dial(ctx, hvsockAddr)
+	if err != nil {
+		logrus.WithError(err).Errorf("error dialing hcsshim external bridge")
+		return
+	}
+
+	// gcs-sidecar can be used for non-confidentail hyperv wcow
+	// as well. So we do not always want to check for initialPolicyStance
+	var initialEnforcer securitypolicy.SecurityPolicyEnforcer
+	if *confidentialWCOW {
```

Comment:
@MahatiC  please let me know if this is ok. gcs-sidecar.exe will be used for non confidential cases as well and therefore we do not want to always check for initial policy